### DISCUSSION TITLE
Set a build format for a specific file(s)

### DISCUSSION THREAD

**Discussion by felixicaza** [withastro/roadmap#350] [Proposal]
Hi there.

I would like to know if it is possible set a file or files have a specific option by the [build.format](https://docs.astro.build/en/reference/configuration-reference/#buildformat), by default all files are generated in directory, but if I set the `build.format` option to `file` **all are generated that way**. Reviewing the code I see that by default the 404 and 500 pages are not affected by these options:

[https://github.com/withastro/astro/blob/main/packages/astro/src/core/build/common.ts#L6](https://github.com/withastro/astro/blob/main/packages/astro/src/core/build/common.ts#L6)

```typescript
const STATUS_CODE_PAGES = new Set(['/404', '/500']);
```

 For example an `404.astro` with the `directory` option it does not generate `/404/index.html`.

This would be very useful to generate other types of custom server error pages or simply to generate one page as a file and the others as directories.

**Comment by fflaten** [comment]
A way to override per page would likely also enable https://github.com/withastro/astro/issues/1582 for SSG without the risk of breaking ex. `/postsbyid/404` for everyone else. The new prerender API uses a exported const for this.

**Comment by florian-lefebvre** [comment]
I think this is now solved by `build.format: 'preserve'` sinces it gives you more control: https://docs.astro.build/en/reference/configuration-reference/#buildformat
